src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs

```diff
@@ -75,14 +80,18 @@ public async Task<HttpResponseMessage> GetResponseByIdAsync(string requestId)
             if (jBatchResponseObject == null)
                 return null;
 
-            JObject jResponseItem = null;
-
-            if (jBatchResponseObject.TryGetValue(CoreConstants.BatchRequest.Responses, out JToken jResponses))
+            if (jBatchResponseObject.RootElement.TryGetProperty(CoreConstants.BatchRequest.Responses, out JsonElement jResponses))
```

Comment:
```suggestion
            if (jBatchResponseObject.RootElement.TryGetProperty(CoreConstants.BatchRequest.Responses, out JsonElement jResponses) && jResponses != null)
```

Reply:
JsonElement is actually a struct and is therefore not nullable. Will resolve with the check jResponses.ValueKind == JsonValueKind.Array